City-of-Helsinki/drupal-helfi-platform-config
# UHF-4423 Translations overhaul

# Translations overhaul [UHF-4423](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4423)

There are too many ways to import translations and some of the translations get overridden even when it shouldn't happen. Within this update we're trying to fix the problems and describe how we should translate content in the future.

## What was done
* Added documentation how to use translations in Helfi context.
* Created 'override' and 'new' folders for custom PO files.
* Updated translation for Generate automatic URL alias.
* Reordered translations and moved them away from overridden translations.

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* **Important** Run `make shell`
   * In shell type: `drush sqlq "select count(*) from locales_target where customized=1"` and make a note of the number of rows found. 
* Update the Helfi Platform config and necessary modules/themes
    * `composer require drupal/helfi_platform_config:dev-UHF-4423_translations_overhaul && composer require drupal/hdbt:dev-UHF-4423_translations_overhaul && composer require drupal/hdbt_admin:dev-UHF-4423_translations_overhaul`
* Run `make drush-updb drush-cr drush-locale-update`

## How to test
* Start with reading the documentation: https://github.com/City-of-Helsinki/drupal-helfi-platform-config/blob/UHF-4423_translations_overhaul/documentation/translations.md
* Go to your local instance and check that 
   * Translations still work as before
   * The paragraphs are translated when creating / editing content with Finnish UI language. =No more "Lisää Text". 
   * _Generate automatic URL alias_ and _Uncheck this to create a custom alias below._  are translated as it is stated in [Aleksi's excel](https://helsinginkaupunki-my.sharepoint.com/:x:/g/personal/aleksi_salonen_hel_fi/EVGs4asmBxdIpGrQrjVcYH0BaLSzARPWyH-unZP68PtTpg)
   * In shell when you run `drush sqlq "select count(*) from locales_target where customized=1"` the amount of customized translations should be significantly less than before the updates. Something like 19 rows.
  
## Other PRs
* https://github.com/City-of-Helsinki/drupal-hdbt/pull/266
* https://github.com/City-of-Helsinki/drupal-hdbt-admin/pull/119
* https://github.com/City-of-Helsinki/drupal-module-helfi-api-base/pull/59
